tsconfig.json

```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    // Target es5 for syntax (IE 11), but expect Vue projects (i.e. Babel) to provide polyfills up to at least es2016
+		// See https://stackoverflow.com/a/57474312/4167438 for explanation
+		"target": "es5",
+		"lib": [
+			"ES2016",
+			"DOM"
+		],
+    "module": "ESNext",
+    "strict": true,
+    "noImplicitAny": true,
```

Comment:
Please don't do this. While I can understand a few exported types in a 3rd party library as a vanity feature, this is overkill and tends to get in the way and cause unnecessary bloat for the sake of making a compiler happy.

Reply:
For sure, happy to - I think in this case, the lib is small enough that it feels like, if we can have full control over this, why don't we? In that I haven't run into anything yet in this codebase where it's *frustrating* to have to define arg types, for example. Whereas I think that happens *a lot* in our larger codebases where we're consuming a lot of untyped dependencies.

In this case, there are 0 real dependencies! 😀 Just `vue` as a peerDep

Reply:
I would argue the exact opposite. "In this case, the lib is small enough that it feels like, if we can have full control over this..." _why would we need to_? Again this goes with the principle of least restriction first mentioned in [my other comment](url).